### DISCUSSION TITLE
The behavior differs depending on the `--net` option.

### DISCUSSION THREAD

**Discussion by atinfinity** [eProsima/Fast-DDS#5100] [Support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I set to `FASTDDS_BUILTIN_TRANSPORTS=DEFAULT` on docker container.
But, the behavior differs depending on the `--net` option.

#### option1: --net=bridge

|FASTDDS_BUILTIN_TRANSPORTS|result|
|---|---|
|DEFAULT|OK|
|UDPv4|OK|
|SHM|NG|

#### option2: --net=host

|FASTDDS_BUILTIN_TRANSPORTS|result|
|---|---|
|DEFAULT|NG|
|UDPv4|OK|
|SHM|NG|

##### option1: --net=bridge

```shell
docker run --rm -it \
           ros:humble-ros-core \
           bash
```

##### option2: --net=host

```shell
docker run --rm -it \
           --net=host \
           ros:humble-ros-core \
           bash
```

### pub

```shell
ros2 topic pub --rate 1 /turtle1/cmd_vel geometry_msgs/msg/Twist "{linear: {x: 2.0, y: 0.0, z: 0.0}, angular: {x: 0.0, y: 0.0, z: 1.8}}"
```

### sub

```shell
ros2 topic echo /turtle1/cmd_vel
```

### Fast DDS version/commit

- ros-humble-fastrtps 2.6.8
- ros-humble-rmw-fastrtps-cpp 6.2.6
- ros-humble-rmw-fastrtps-shared-cpp 6.2.6

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM, UDPv4, Shared Memory Transport (SHM)

### Additional context

- Host OS: Ubuntu 22.04(x86_64)

**Comment by JesusPoderoso** [comment]
Hi @atinfinity, thanks for using Fast DDS.
Please, refer to [Fast DDS docker deployments](https://fast-dds.docs.eprosima.com/en/latest/docker/shm_docker.html) documentation section.
I am moving this issue to the Support discussion forum according to the [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md).
